guides/advancingGuides/advanceGuideUsingDynamicPermalinkButton/UsingClickableImage/js_for_clickable_image.js

```diff
@@ -0,0 +1,29 @@
+(function wireNextGuideButton(step) {
+    step && step.attachEvent(step.guideElement[0], 'click', function(e) {
+        
+        // get the first clickable image in the guide.
+        var advanceButton = pendo.dom(e.target || e.srcElement).closest('.opt1');
+
+        if (advanceButton.length) {
+            // get the text value from the metadata to be used ahead in the URL.
+            var account_name = document.getElementsByClassName("_pendo-hidden-account_")[0].innerHTML
+
+            // build your URL by using the HTML variable above.
+            var newurl = "https://" + account_name + ".app.com/settings/manage?pendo=guide_id";
```

Comment:
There's actually much easier way to do this in Pendo that doesn't require sourcing the attributes via HTML elements.  

You can use the `pendo.gerSerializedMetadata()` function on the Pendo Agent to retrieve all currently present metadata values in JSON format.  For example:

`pendo.getSerializedMetadata().visitor.id` will return the active visitorID.  `pendo.getSerializedMetadata().visitor.role` will return the role metadata field off of the visitor entity.

Rather than wiring it up this way, can you refactor to use this?  I think there's value to the dynamic link snippet, but I'd rather use the simpler approach.

Reply:
Looking at your comments, actually, it seems like you know about pendo.getSerializedMetadata() -- if you think there's a reason why pulling innerText off of HTML elements is superior to a pure JS approach, let me know.  Otherwise, my personal take is that just constructing a URL with string interpolation this way is easier.

Reply:
Makes sense @fortynachos I will add comments to provide both the methods as pendo.getSerializedMetadata has clear advantages here.